airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -699,6 +700,36 @@ class FormStream(Stream):
     created_at_field = "createdAt"
 
 
+class FormSubmissionStream(Stream):
+    """Marketing Forms, API v1
+    This endpoint requires the forms scope.
+    Docs: https://legacydocs.hubspot.com/docs/methods/forms/get-submissions-for-a-form
+    """
+
+    url = "/form-integrations/v1/submissions/forms"
+    limit = 50
+    updated_at_field = "updatedAt"
+
+    def _transform(self, records: Iterable) -> Iterable:
+        for record in super()._transform(records):
+            keys = record.keys()
+            if "id" not in keys:
+                record["id"] = uuid4()
+            if "updatedAt" not in keys:
+                record["updatedAt"] = record["submittedAt"]
+
+            yield record
+
+    def list(self, fields) -> Iterable:
+        for form in self.read(getter=partial(self._api.get, url="/marketing/v3/forms")):
+            for submission in self.read(getter=partial(self._api.get, url=f"{self.url}/{form['id']}")):
+                del submission["id"]
+                del submission["updatedAt"]
```

Comment:
same reason it's not common to delete data during a ELT process, can you check the other streams from Hubspot and try to leverage to the one you're implementing. If you need a different function please add comments describing the reason why you're doing it so others in the future will understand easier

Reply:
`id` was added temporarily because of the above mentioned reasons, and `updatedAt` is copied from `submittedAt` because for fetching the Forms a different updated_at_field is needed (`updatedAt`) than for the submissions (`submittedAt`) (which is used here later: https://github.com/airbytehq/airbyte/blob/0eba52e002235caab14cdbb8f4c5847f8b8b0930/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py#L319).
Unfortunately in the class only one can be defined and in this case I needed to use 2 separate endpoints with different returned timestamps. Basically I wanted to use the solution what I saw here:
https://github.com/airbytehq/airbyte/blob/0eba52e002235caab14cdbb8f4c5847f8b8b0930/airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py#L579
but it couldn't be applied as is.